### PR TITLE
Use disk as boot device when booting from PXE-boot configured VM seco…

### PR DESCRIPTION
**Pull request by rawagner**
…nd time

- [x] Depends on https://github.com/kubevirt/web-ui-components/pull/66

### CODE REVIEW

File: frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import React from 'react';
+import * as _ from 'lodash';
 
 import { PromiseComponent } from '../utils/okdutils';
 import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/okdfactory';
-import { k8sPatch } from '../../../module/k8s';
+import { k8sPatch, k8sUpdate } from '../../../module/k8s';
+import { ANNOTATION_FIRST_BOOT } from 'kubevirt-web-ui-components/dist/js';
```
Comment (mareklibra) [inline frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx]: We should avoid the `dist/js` in the import.
Can you please verify, if https://github.com/kubevirt/web-ui-components/pull/68 fixes that?

Same occurrence is in `vm.jsx`.

File: frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx [hunk 2]
```diff
@@ -13,12 +15,36 @@ class StartStopVmModal extends PromiseComponent {
...
   submit(event) {
     event.preventDefault();
-    const patch = [{
-      op: 'replace',
-      path: '/spec/running',
-      value: this.props.start
-    }];
-    const promise = k8sPatch(this.props.kind, this.props.resource, patch);
+    let promise;
+    const vm = this.props.resource;
```
Comment (mareklibra) [inline frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx]: We should start using `propTypes` and ideally even `defaultProps`. WDYT?
If you agree, do you prefer to add it within this or a follow-up PR?

File: frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx [hunk 3]
```diff
@@ -13,12 +15,36 @@ class StartStopVmModal extends PromiseComponent {
...
+
+    // handle PXE boot
+    const annotations = _.get(vm, 'metadata.annotations', {});
+    if (this.props.start && annotations[ANNOTATION_FIRST_BOOT]) {
+      if (annotations[ANNOTATION_FIRST_BOOT] === 'true'){
+        annotations[ANNOTATION_FIRST_BOOT] = 'false';
+      } else {
+        // find bootable disk and change boot order
+        const disks = _.get(vm, 'spec.template.spec.domain.devices.disks', []);
+        const bootableDisk = disks.find(disk => disk.bootOrder === 2);
```
Comment (mareklibra) [inline frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx]: Should not happen, but still possible: the `bootableDisk` can be na array of >1 elements

File: frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx (hunk 1 above)
Comment (rawagner) [reply frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx]: that helps!

File: frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx (hunk 2 above)
Comment (rawagner) [reply frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx]: I will add propTypes within this PR

**Pull request merged**